alf/environments/metadrive/agent_perception.py

```diff
@@ -0,0 +1,234 @@
+# Copyright (c) 2022 Horizon Robotics and ALF Contributors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple, Optional, NamedTuple
+import functools
+
+import torch
+import numpy as np
+
+from alf.tensor_specs import TensorSpec
+from .geometry import FieldOfView, Polyline
+
+try:
+    import metadrive
+    from metadrive.component.vehicle.base_vehicle import BaseVehicle
+    from metadrive.engine.base_engine import BaseEngine
+except ImportError:
+    from unittest.mock import Mock
+    # create 'metadrive' as a mock to not break python argument type hints
+    metadrive = Mock()
+
+
+class AgentPerception(object):
```

Comment:
To me, the name AgentPerception could be changed to something else (e. g. DynamicObjectPerception) better reflecting its intended meaning of perceiving other agent. Currently, from the name only, it has some confusion with the Perception module of an ego agent.

Reply:
Noted. This actually belongs to a previous PR. I can make another one for the name chaning.